lang/cs/Org.Apache.REEF.Network.Tests/GroupCommunication/GroupCommunicationTests.cs

```diff
@@ -745,9 +746,17 @@ public static List<ICommunicationGroupClient> CommGroupClients(string groupName,
         public static NetworkService<GroupCommunicationMessage> BuildNetworkService(
             IPEndPoint nameServerEndpoint, IObserver<NsMessage<GroupCommunicationMessage>> handler)
         {
-            var remoteManagerFactory = TangFactory.GetTang().NewInjector().GetInstance<IRemoteManagerFactory>();
+            var injector = TangFactory.GetTang().NewInjector();
+            var remoteManagerFactory = injector.GetInstance<IRemoteManagerFactory>();
+            var tcpPortProviderFactory = injector.GetInstance<ITcpPortProviderFactory>();
+            var tcpPortProvider = tcpPortProviderFactory.GetInstance();
             return new NetworkService<GroupCommunicationMessage>(
-                0, handler, new StringIdentifierFactory(), new GroupCommunicationMessageCodec(), new NameClient(nameServerEndpoint.Address.ToString(), nameServerEndpoint.Port), remoteManagerFactory);
+                0, handler, new StringIdentifierFactory(), 
+                new GroupCommunicationMessageCodec(), 
+                new NameClient(nameServerEndpoint.Address.ToString(), 
+                    nameServerEndpoint.Port), 
+                    remoteManagerFactory,
```

Comment:
Wouldn't it be easier to pass the `tcpPortProvider` straight into `RemoteManagerFactory`?